typescript-generator-core/src/main/java/cz/habarta/typescript/generator/parser/DeprecationEnricher.java

```diff
@@ -40,46 +45,90 @@ private PropertyModel enrichProperty(PropertyModel property) {
         final List<EnumMemberModel> members = mapList(enumModel.getMembers(), enumMember -> enrichEnumMember(enumMember));
         return enumModel
                 .withMembers(members)
                 .withComments(addDeprecation(enumModel.getComments(), enumModel.getOrigin()));
     }
 
-    private EnumMemberModel enrichEnumMember(EnumMemberModel enumMember) {
+    public EnumMemberModel enrichEnumMember(EnumMemberModel enumMember) {
         return enumMember
                 .withComments(addDeprecation(enumMember.getComments(), enumMember.getOriginalField()));
     }
 
-    private RestApplicationModel enrichRestApplication(RestApplicationModel restApplicationModel) {
+    public RestApplicationModel enrichRestApplication(RestApplicationModel restApplicationModel) {
         final List<RestMethodModel> enrichedRestMethods = mapList(restApplicationModel.getMethods(), restMethod -> enrichRestMethod(restMethod));
         return restApplicationModel.withMethods(enrichedRestMethods);
     }
 
-    private RestMethodModel enrichRestMethod(RestMethodModel method) {
+    public RestMethodModel enrichRestMethod(RestMethodModel method) {
         return method
                 .withComments(addDeprecation(method.getComments(), method.getOriginalMethod()));
     }
 
-    private static <R, T> List<R> mapList(List<T> list, Function<T, R> mapper) {
-        return list.stream().map(mapper).collect(Collectors.toList());
+    public List<String> addDeprecation(List<String> comments, AnnotatedElement annotatedElement) {
+        if (annotatedElement == null || !annotatedElement.isAnnotationPresent(Deprecated.class) || containsDeprecatedTag(comments)) {
+            return comments;
+        }
+
+        String deprecatedComment = convertToComment(annotatedElement.getAnnotation(Deprecated.class));
+        return Utils.concat(comments, Collections.singletonList(deprecatedComment));
+    }
+
+    protected String convertToComment(Deprecated deprecated) {
+        // support for java 9+ syntax
+        String since = getSince(deprecated);
+        boolean forRemoval = getForRemoval(deprecated);
+
+        List<String> additional = new ArrayList<>();
+        if (!"".equals(since)) {
+            additional.add("since: " + since);
+        }
+        if (forRemoval) {
+            additional.add("forRemoval: true");
+        }
+        return additional.isEmpty() ? DEPRECATED : (DEPRECATED + " " + String.join("; ", additional));
+    }
+
+    protected String getSince(Deprecated deprecated) {
+        return Arrays.stream(deprecated.getClass().getMethods())
+            .filter(m -> m.getName().equals("since"))
+            .map(m -> {
+                try {
+                    return (String) m.invoke(deprecated);
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    return "";
+                }
+            })
+            .findFirst().orElse("");
     }
```

Comment:
For getting annotation elements using reflections there is a method [getAnnotationElementValue](https://github.com/vojtechhabarta/typescript-generator/blob/95a9bf5fc16056c507dcf330584b46a5b0be2bf1/typescript-generator-core/src/main/java/cz/habarta/typescript/generator/util/Utils.java#L194).